### PR TITLE
Preferences improvements

### PR DESCRIPTION
**Pull request by paulovmr**
GUVNOR-2769: [Preferences] Add helper icons to every property in preferences pages
GUVNOR-2796: [Preferences] Exception is thrown when Organization Unit Identifier property is empty
GUVNOR-2795: [Preferences] ProjectBuildingException is thrown when Project Group ID or Project Version properties contains empty inputs
GUVNOR-2770: [Preferences] Repository Default Scheme property has to be always 'git'
GUVNOR-2763: [Preferences] Project Default Branch change causes error

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:
* https://github.com/AppFormer/uberfire/pull/754
* https://github.com/kiegroup/guvnor/pull/466
* https://github.com/kiegroup/kie-wb-common/pull/944

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties [hunk 1]
```diff
@@ -18,21 +18,31 @@ LibraryPreferences.Label=Library Preferences
...
+LibraryOrganizationalUnitPreferences.AliasInPlural.Help=Customized alias (in plural) to be used instead of teams.
 LibraryRepositoryPreferences.Label=Default Repository fields
 LibraryRepositoryPreferences.Name=Name
+LibraryRepositoryPreferences.Name.Help=Name of the repository that will be created automatically if no repository exists inside of the selected team.
 LibraryRepositoryPreferences.Scheme=Scheme
+LibraryRepositoryPreferences.Scheme.Help=Scheme of the repository that will be created automatically if no repository exists inside of the selected team.
 LibraryProjectPreferences.Label=Default Project fields
-LibraryProjectPreferences.GroupId=Group ID
 LibraryProjectPreferences.Version=Version
+LibraryProjectPreferences.Version.Help=Default version to be used when quickly creating a project.
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: 'when quickly creating a project..'  - this looks as if "user's speed" in creating new project played some role :)

How about 'when creating project using `Quick setup` option'

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties [hunk 2]
```diff
@@ -18,21 +18,31 @@ LibraryPreferences.Label=Library Preferences
...
 LibraryProjectPreferences.Description=Description
+LibraryProjectPreferences.Description.Help=Default description to be loaded when quickly creating a project.
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: The same as above

**Comment by jhrcek** [comment]
Jenkins execute full downstream build

@tomasdavidorg could you please test this locally and provide some feedback?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryOrganizationalUnitPreferences.java [hunk 3]
```diff
@@ -19,24 +19,33 @@
...
 import org.uberfire.preferences.shared.annotations.WorkbenchPreference;
 import org.uberfire.preferences.shared.bean.BasePreference;
+import org.uberfire.preferences.shared.impl.validation.NotEmptyValidator;
 
 @WorkbenchPreference(identifier = "LibraryOrganizationalUnitPreferences",
         bundleKey = "LibraryOrganizationalUnitPreferences.Label")
 public class LibraryOrganizationalUnitPreferences implements BasePreference<LibraryOrganizationalUnitPreferences> {
 
-    @Property(bundleKey = "LibraryOrganizationalUnitPreferences.Name")
+    @Property(bundleKey = "LibraryOrganizationalUnitPreferences.Name",
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryOrganizationalUnitPreferences.java]: Formatting?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryPreferences.java [hunk 4]
```diff
@@ -33,7 +33,8 @@
     @Property(bundleKey = "LibraryPreferences.ProjectPreferences")
     LibraryProjectPreferences projectPreferences;
 
-    @Property(bundleKey = "LibraryPreferences.ImportProjectsUrl")
+    @Property(bundleKey = "LibraryPreferences.ImportProjectsUrl",
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryPreferences.java]: Formatting?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryProjectPreferences.java [hunk 5]
```diff
@@ -16,21 +16,28 @@
...
 import org.uberfire.preferences.shared.annotations.WorkbenchPreference;
 import org.uberfire.preferences.shared.bean.BasePreference;
+import org.uberfire.preferences.shared.impl.validation.NotEmptyValidator;
 
 @WorkbenchPreference(identifier = "LibraryProjectPreferences",
         bundleKey = "LibraryProjectPreferences.Label")
 public class LibraryProjectPreferences implements BasePreference<LibraryProjectPreferences> {
 
-    @Property(bundleKey = "LibraryProjectPreferences.Version")
+    @Property(bundleKey = "LibraryProjectPreferences.Version",
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryProjectPreferences.java]: Formatting?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryRepositoryPreferences.java [hunk 6]
```diff
@@ -16,18 +16,24 @@
...
 import org.uberfire.preferences.shared.annotations.WorkbenchPreference;
 import org.uberfire.preferences.shared.bean.BasePreference;
+import org.uberfire.preferences.shared.impl.validation.NotEmptyValidator;
 
 @WorkbenchPreference(identifier = "LibraryRepositoryPreferences",
         bundleKey = "LibraryRepositoryPreferences.Label")
 public class LibraryRepositoryPreferences implements BasePreference<LibraryRepositoryPreferences> {
 
-    @Property(bundleKey = "LibraryRepositoryPreferences.Name")
+    @Property(bundleKey = "LibraryRepositoryPreferences.Name",
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryRepositoryPreferences.java]: Formatting?

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties (hunk 1 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: Done.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties (hunk 2 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/resources/org/kie/workbench/common/screens/library/client/resources/i18n/Library.properties]: Done.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryOrganizationalUnitPreferences.java (hunk 3 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryOrganizationalUnitPreferences.java]: @manstis Actually I formatted the file, unfortunately this is how annotation members get formatted...

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryPreferences.java (hunk 4 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryPreferences.java]: @manstis Actually I formatted the file, unfortunately this is how annotation members get formatted...

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryProjectPreferences.java (hunk 5 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryProjectPreferences.java]: @manstis Actually I formatted the file, unfortunately this is how annotation members get formatted...

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryRepositoryPreferences.java (hunk 6 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/main/java/org/kie/workbench/common/screens/library/api/preferences/LibraryRepositoryPreferences.java]: @manstis Actually I formatted the file, unfortunately this is how annotation members get formatted...

**Comment by ederign** [comment]
@jhrcek @tomasdavidorg can you guys please review this?

**Comment by manstis** [comment]
@jhrcek @tomasdavidorg Hi guys, please let me know when you're OK these PRs being merged.

**Comment by tomasdavidorg** [comment]
@manstis Hi I am just looking at this. I will let you know.

**Comment by manstis** [comment]
@tomasdavidorg Thanks. @jhrcek Are you OK with the merge now?

**Pull request merged**